src/Voron/Data/BTrees/Tree.Stream.cs

```diff
@@ -201,7 +201,9 @@ private void AllocateNextPage()
 
                 _currentPage = nextPage;
                 _currentPage.Flags |= PageFlags.Stream;
-                _parent.State.OverflowPages += _numberOfPagesPerChunk;
+
+                ref var state = ref _parent.State.Modify();
```

Comment:
Previously we've not set state as modified, this is a fix now?

Reply:
It was already set as modified before entering here. This is what I mean about being more explicit.